db/repos/transactions.js

```diff
@@ -25,83 +26,244 @@ var columnSet;
  * @param {Database} db - Instance of database object from pg-promise
  * @param {Object} pgp - pg-promise instance to utilize helpers
  * @constructor
- * @return {TransactionsRepo}
+ * @return {TransactionsRepository}
  */
-function TransactionsRepo(db, pgp) {
-	this.db = db;
-	this.pgp = pgp;
-
-	this.sortFields = [
-		'id',
-		'blockId',
-		'amount',
-		'fee',
-		'type',
-		'timestamp',
-		'senderPublicKey',
-		'senderId',
-		'recipientId',
-		'confirmations',
-		'height',
-	];
-
-	this.dbTable = 'trs';
-
-	this.dbFields = [
-		'id',
-		'blockId',
-		'type',
-		'timestamp',
-		'senderPublicKey',
-		'requesterPublicKey',
-		'senderId',
-		'recipientId',
-		'amount',
-		'fee',
-		'signature',
-		'signSignature',
-		'signatures',
-	];
-
-	if (!columnSet) {
-		columnSet = {};
-		var table = new pgp.helpers.TableName({
-			table: this.dbTable,
-			schema: 'public',
+class TransactionsRepository {
+	constructor(db, pgp) {
+		this.db = db;
+		this.pgp = pgp;
+
+		this.sortFields = [
+			'id',
+			'blockId',
+			'amount',
+			'fee',
+			'type',
+			'timestamp',
+			'senderPublicKey',
+			'senderId',
+			'recipientId',
+			'confirmations',
+			'height',
+		];
+
+		this.dbTable = 'trs';
+
+		this.dbFields = [
+			'id',
+			'blockId',
+			'type',
+			'timestamp',
+			'senderPublicKey',
+			'requesterPublicKey',
+			'senderId',
+			'recipientId',
+			'amount',
+			'fee',
+			'signature',
+			'signSignature',
+			'signatures',
+		];
+
+		if (!cs.insert) {
+			cs.insert = new pgp.helpers.ColumnSet(this.dbFields, {
+				table: this.dbTable,
+			});
+			cs.insert = cs.insert.merge([{ name: 'recipientId', def: null }]);
+		}
+
+		this.cs = cs;
+
+		this.transactionsRepoMap = {};
+		this.transactionsRepoMap[transactionTypes.SEND] = 'transactions.transfer';
+		this.transactionsRepoMap[transactionTypes.DAPP] = 'transactions.dapp';
+		this.transactionsRepoMap[transactionTypes.DELEGATE] =
+			'transactions.delegate';
+		this.transactionsRepoMap[transactionTypes.IN_TRANSFER] =
+			'transactions.inTransfer';
+		this.transactionsRepoMap[transactionTypes.OUT_TRANSFER] =
+			'transactions.outTransfer';
+		this.transactionsRepoMap[transactionTypes.MULTI] =
+			'transactions.multisignature';
+		this.transactionsRepoMap[transactionTypes.SIGNATURE] =
+			'transactions.signature';
+		this.transactionsRepoMap[transactionTypes.VOTE] = 'transactions.vote';
```

Comment:
This `transactionRepoMap` a static stuff and never change during the instance of the repo. Should not we move it outside constructor?

Reply:
We certainly can move it outside of the constructor. I'm not sure any of it will be left there after the second round of refactoring when the entire logic may change. See the next task on the list: https://github.com/LiskHQ/lisk/issues/1432